### PR TITLE
Modified mask_fit handling in FoVBackgroundMaker

### PR DESCRIPTION
**Pull request by luca-giunti**
**Description**
This pull request addresses #2921: the `mask_fit` was silently overridden by the `FoVBackgroundMaker`. Now it is taken into account.

### CODE REVIEW

File: gammapy/makers/background/tests/test_fov.py [hunk 1]
```diff
@@ -284,3 +284,18 @@ def test_fov_bkg_maker_scale_fail(obs_dataset, exclusion_mask, caplog):
...
+    region = CircleSkyRegion(obs_dataset._geom.center_skydir, Angle(0.4, "deg"))
+    test_dataset.mask_fit = obs_dataset._geom.region_mask(regions=[region])
+
+    dataset = fov_bkg_maker.run(test_dataset)
+
+    model = dataset.models[f"{dataset.name}-bkg"].spectral_model
+
+    assert_allclose(model.norm.value, 0.9975, rtol=1e-3)
+    assert_allclose(model.norm.error, 0.1115, rtol=1e-3)
+    assert_allclose(model.tilt.value, 0.0, rtol=1e-2)
```
Comment (registerrier) [inline gammapy/makers/background/tests/test_fov.py:301]: Maybe add a check that the mask fit is back to its initial value.

**Review by registerrier — APPROVED** [review]
Thanks @luca-giunti . This looks good

See inline comment.

File: gammapy/makers/background/tests/test_fov.py [hunk 2]
```diff
@@ -284,3 +290,20 @@ def test_fov_bkg_maker_scale_fail(obs_dataset, exclusion_mask, caplog):
...
+@requires_data()
+def test_fov_bkg_maker_mask_fit_handling(obs_dataset, exclusion_mask):
+    fov_bkg_maker = FoVBackgroundMaker(method="scale", exclusion_mask=exclusion_mask)
+    test_dataset = obs_dataset.copy(name="test-fov")
+    region = CircleSkyRegion(obs_dataset._geom.center_skydir, Angle(0.4, "deg"))
+    mask_fit = obs_dataset._geom.region_mask(regions=[region])
+    test_dataset.mask_fit = mask_fit
+
+    dataset = fov_bkg_maker.run(test_dataset)
+    assert np.all((test_dataset.mask_fit == mask_fit).data) == True
```
Comment (registerrier) [inline gammapy/makers/background/tests/test_fov.py:304]: `assert_allclose` should better for this

Comment (adonath) [reply gammapy/makers/background/tests/test_fov.py:304]: I think for booleans there should be no uncertainty. However one should simplify the expression to:
```
assert np.all(test_dataset.mask_fit == mask_fit)
```
`Map` objects are interpreted as `np.ndarray` when put into a numpy function...

Comment (registerrier) [reply gammapy/makers/background/tests/test_fov.py:304]: Can you try this solution @luca-giunti ?

Comment (luca-giunti) [reply gammapy/makers/background/tests/test_fov.py]: @registerrier Sure, sorry for the delay

File: gammapy/makers/background/fov.py [hunk 3]
```diff
@@ -193,8 +193,10 @@ def run(self, dataset, observation=None):
 
         """
         mask_fit = dataset.mask_fit
-
-        dataset.mask_fit = self.make_exclusion_mask(dataset)
+        if mask_fit:
+            dataset.mask_fit *= self.make_exclusion_mask(dataset)
+        else:
+            dataset.mask_fit = self.make_exclusion_mask(dataset)
```
Comment (adonath) [inline gammapy/makers/background/fov.py:199]: I noticed this behavior is actually not documented. Could you add one or two sentences in the docstring of `FoVBackgroundMaker` that it modifies the `mask_fit` by combining it with the exclusion mask?

Comment (adonath) [inline gammapy/makers/background/fov.py:199]: I noticed this behavior is actually not documented. Could you add one or two sentences in the docstring of `FoVBackgroundMaker` that it modifies the `mask_fit` by combining it with the exclusion mask?

**Review by adonath — COMMENTED** [review]
Thanks @luca-giunti, I left one comment on documentation. Otherwise looks good to me...

**Comment by luca-giunti** [comment]
@adonath done!

**Review by adonath — APPROVED** [review]
Thanks a lot @luca-giunti, no further comments from my side!

**Pull request merged**